apps/judicial-system/backend/src/app/modules/case/case.service.ts

```diff
@@ -959,33 +974,37 @@ export class CaseService {
           theCase.state,
         )
       } else if (completedCaseStates.includes(updatedCase.state)) {
-        if (isIndictmentCase(updatedCase.type)) {
+        if (isIndictment) {
           await this.addMessagesForCompletedIndictmentCaseToQueue(
             updatedCase,
             user,
           )
         } else {
           await this.addMessagesForCompletedCaseToQueue(updatedCase, user)
         }
+      } else if (updatedCase.state === CaseState.SUBMITTED && isIndictment) {
```

Comment:
I'd like to refactor this when we have time, all the if cases are getting a bit hard to read.

Reply:
Agree. We should also speed this up by waiting in parallel wherever posible.